### ISSUE TITLE
some question about k_inv_dot_xy1?

### ISSUE THREAD

**Issue by XGuider** [svip-lab/PlanarReconstruction#23]
thnx your best work ,but i But I don’t understand k_inv_dot_xy1 well.
`    x = torch.arange(w, dtype=torch.float32).view(1, w) / w * 640
    y = torch.arange(h, dtype=torch.float32).view(h, 1) / h * 480

    x = x.to(device)
    y = y.to(device)
    xx = x.repeat(h, 1)
    yy = y.repeat(1, w)
    xy1 = torch.stack((xx, yy, torch.ones((h, w), dtype=torch.float32).to(device)))  # (3, h, w)
    xy1 = xy1.view(3, -1)  # (3, h*w)
`
why do this?what's function?thnx

**Comment by niujinshuchong** [comment]
k_inv_dot_xy1 is used to convert the depth to point cloud. Because we use the same intrinsic matrix for all the images, pre computing the value of k_inv_dot_xy1 will save some computation.

**Comment by XGuider** [comment]
yes, convert the depth to point cloud like this:depth*K_inv*[u,v,1].so
 `x = torch.arange(w, dtype=torch.float32).view(1, w) / w * 640
y = torch.arange(h, dtype=torch.float32).view(h, 1) / h * 480`
Does this step initialize u,v?and why?
question2:
`depth_map = 1. / np.sum(self.K_inv_dot_xy_1.reshape(3, -1) * plane_parameters.reshape(3, -1), axis=0)`
this code confused me,===> (1/D)[X,Y,D]*N/D
and (n/d)*Q=1 .n is the normal vector,why not (n/d)*Q =0
Thank you for your prompt reply..

**Comment by niujinshuchong** [comment]
Yes. x and y here is the uv coords. 

 n * Q = 1 is the plane equation. To compute the depth from plane parameter n, please see Eq.2 in this paper: http://openaccess.thecvf.com/content_ECCV_2018/papers/Fengting_Yang_Recovering_3D_Planes_ECCV_2018_paper.pdf.

**Comment by XGuider** [comment]
okay,i get.thnx so much.
